components/smalltalk/squeak5/squeak5.p5m

```diff
@@ -0,0 +1,53 @@
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2020 David Stes
+#
+
+set name=pkg.fmri value=pkg:/$(COMPONENT_FMRI)@$(IPS_COMPONENT_VERSION),$(BUILD_VERSION)
+set name=pkg.summary value="$(COMPONENT_SUMMARY)"
+set name=info.classification value="$(COMPONENT_CLASSIFICATION)"
+set name=info.upstream-url value=$(COMPONENT_PROJECT_URL)
+set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
+
+license $(COMPONENT_LICENSE_FILE) license='$(COMPONENT_LICENSE)'
+
+# see desktop/rss-glx/rss-glx.p5m:
+<transform file -> add pkg.depend.bypass-generate libGL\.so\.1>
+<transform file -> add pkg.depend.bypass-generate vm-display-X11\.so>
+
+file mode=0555 path=usr/bin/squeak
+file path=usr/doc/squeak/COPYING.gz
+file path=usr/doc/squeak/COPYRIGHT.gz
+file path=usr/doc/squeak/LICENSE.gz
+file path=usr/doc/squeak/README.Contributing.gz
+file path=usr/doc/squeak/README.Keyboard.gz
+file path=usr/doc/squeak/README.Sound.gz
+file path=usr/lib/$(MACH64)/squeak/5.0-202004291145/B3DAcceleratorPlugin.so
+file path=usr/lib/$(MACH64)/squeak/5.0-202004291145/FileAttributesPlugin.so
+file path=usr/lib/$(MACH64)/squeak/5.0-202004291145/ImmX11Plugin.so
+file path=usr/lib/$(MACH64)/squeak/5.0-202004291145/LocalePlugin.so
+file path=usr/lib/$(MACH64)/squeak/5.0-202004291145/Squeak3D.so
+file path=usr/lib/$(MACH64)/squeak/5.0-202004291145/SqueakFFIPrims.so
+file path=usr/lib/$(MACH64)/squeak/5.0-202004291145/SqueakSSL.so
+file path=usr/lib/$(MACH64)/squeak/5.0-202004291145/UUIDPlugin.so
+file path=usr/lib/$(MACH64)/squeak/5.0-202004291145/UnicodePlugin.so
+file path=usr/lib/$(MACH64)/squeak/5.0-202004291145/UnixOSProcessPlugin.so
+file path=usr/lib/$(MACH64)/squeak/5.0-202004291145/XDisplayControlPlugin.so
+file mode=0555 path=usr/lib/$(MACH64)/squeak/5.0-202004291145/squeak
+file path=usr/lib/$(MACH64)/squeak/5.0-202004291145/vm-display-X11.so
+file path=usr/lib/$(MACH64)/squeak/5.0-202004291145/vm-display-null.so
+file path=usr/lib/$(MACH64)/squeak/5.0-202004291145/vm-sound-null.so
+file path=usr/lib/$(MACH64)/squeak/5.0-202004291145/vm-sound-pulse.so
+file path=usr/share/man/man1/inisqueak.1
```

Comment:
When I run gmake sample-manifest I'll get more libraries:
╰─➤  diff sample-manifest.p5m sample-manifest.p5m.orig
59,62d58
< file path=usr/lib/$(MACH64)/squeak/5.0-202004291145/vm-sound-OSS.a
< file path=usr/lib/$(MACH64)/squeak/5.0-202004291145/vm-sound-OSS.so
< file path=usr/lib/$(MACH64)/squeak/5.0-202004291145/vm-sound-Sun.a
< file path=usr/lib/$(MACH64)/squeak/5.0-202004291145/vm-sound-Sun.so

Do you have the audio headers installed?
─➤  pkg list header-audio
NAME (PUBLISHER)                                  VERSION                    IFO
system/header/header-audio                        0.5.11-2020.0.1.19654      i--

Reply:
You have a build server that has more packages than my system, I think.  What happens is, the "configure" script detects some audio headers and then enables the build of ALSA, OSS, Sun etc. depending on the header files it can find.   The configure script has an option -without-OSS but unfortunately it does not have an option --without-Sun so when it finds the sun audio header, it builds that plugin (which worked fine in your case but I suspect the plugin compiles, but does not work, it's old SPARC Sun audio I think).   Note that my personal preference would be to ship only the vm-sound-pulse.so module.   The squeakvm will check for modules in the following order: OSS, Sun, pulse.   This order is the reverse of what I want.   I want it to check first for pulse, then for OSS, then for Sun audio.    But by having only the vm-sound-pulse.so in the manifest, and only packaging pulse,  the problem is solved because the squeakvm will only find the pulse plugin so it will use that, which is fine.   If I'd package OSS, Sun and pulse plugins, then the "search order" of the squeakvm is doing the "wrong thing".   Wrong in the sense of not using the pulse-server unless the VM is started with the -vm-sound-pulse command line switch.   This is something that most users will not know about, in fact, it's something I discovered by experimenting a bit.   Note that on some other Linux distributions they also use "pulse audio" and some Squeak users also would like the search order to be changed so that it uses pulse if it can find it, but the VM is not doing that at this moment, and some Linux distributions that do not use pulseaudio object to the change.

Reply:
Valid reasoning, but you should have the libs in the sample-manifest because it's not only the base for the package's p5m file but also let you (or another developer) know whether something has changed during an update of the package. If there is something missing now it might be visible next time and then included into squeak5.p5m creating all the hassle you describe.

Reply:
So basically short story : it is intentional that there is only vm-sound-pulse in the manifests of squeak4 and squeak5.  I'm not going to change that.